kubernetes/client/kubernetes_test.go

```diff
@@ -0,0 +1,277 @@
+			tempMountPath:   "/wr-tmp/",
+			cmdArgs:         []string{"tail", "-f", "/dev/null"},
+			configMountPath: "/scripts/",
+			configMapData:   "echo \"hello world\"",
+			resourceReq: apiv1.ResourceRequirements{
+				Requests: apiv1.ResourceList{
+					apiv1.ResourceCPU:              *resource.NewMilliQuantity(int64(1)*1000, resource.DecimalSI),
+					apiv1.ResourceMemory:           *resource.NewQuantity(int64(1)*1024*1024, resource.BinarySI),
+					apiv1.ResourceEphemeralStorage: *resource.NewQuantity(int64(0)*1024*1024*1024, resource.BinarySI),
+				},
+				Limits: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewMilliQuantity(int64(1)*1000, resource.DecimalSI),
+					apiv1.ResourceMemory: *resource.NewQuantity(int64(1)*1024*1024, resource.BinarySI),
+				},
+			},
+		},
+	}
+	for _, c := range cases {
+
+		// Test the creation of config maps (2 birds one stone) We won't delete
+		// this now so we can use it later.
+		configmap, err := tc.CreateInitScriptConfigMap(c.configMapData)
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		expectedData := "#!/usr/bin/env bash\nset -euo pipefail\necho \"Running init script\"" +
+			"\necho \"hello world\"\necho \"Init Script complete, executing arguments provided\"\nexec $@"
+
+		if configmap.Data[client.DefaultScriptName] != expectedData {
+			t.Error(fmt.Errorf("Unexpected contents of config map, got:\n%s \nexpect:\n%s", configmap.Data[client.DefaultScriptName], expectedData))
+		}
+
+		// create spawn request we run the init script created from wherever
+		// we've decided to mount it.
+
+		pod, err := tc.Spawn(c.containerImage, c.tempMountPath,
+			c.configMountPath+client.DefaultScriptName,
+			c.cmdArgs, configmap.ObjectMeta.Name,
+			c.configMountPath, c.resourceReq)
+		if err != nil {
+			t.Error(err.Error())
+		}
+
+		// wait on the status to be running or pending
+		err = wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+			pod, err = clientset.CoreV1().Pods(testingNamespace).Get(pod.ObjectMeta.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			if pod.Status.Phase == "Pending" || pod.Status.Phase == "Running" {
+				return true, nil
+			}
+			return false, nil
+		})
+
+		if err != nil {
+			t.Error(fmt.Errorf("Spawn failed: %s", err))
+		}
+
+		// Now delete it
+		err = tc.DestroyPod(pod.ObjectMeta.Name)
+		if err != nil {
+			t.Error(fmt.Errorf("Deleting pod failed: %s", err))
+		}
+
+	}
+
+}
+
+func TestInWrPod(t *testing.T) {
+	// These tests are not run inside a wr pod. Expect false
+	inwr := client.InWRPod()
+	if inwr {
+		t.Error("This test is not run inside a pod wr controlls, should return false.")
+	}
+}
+
+func TestCreateInitScriptConfigMapFromFile(t *testing.T) {
+	cases := []struct {
+		fileData string
+		filePath string
+	}{
+		{
+			fileData: "echo \"hello world\"",
+			filePath: "/tmp/d1",
+		},
+	}
+
+	for _, c := range cases {
+		err := ioutil.WriteFile(c.filePath, []byte(c.fileData), 0644)
```

Comment:
You don't defer deletion of this file. Also, rather that put in in /tmp, best to use a https://golang.org/pkg/io/ioutil/#TempDir (which you still need to defer removal of).